app/grandchallenge/core/templatetags/bleach.py

```diff
@@ -6,18 +8,45 @@
 from markdown import markdown as render_markdown
 from markdown.extensions.toc import TocExtension
 
-from grandchallenge.core.utils.markdown import LinkBlankTargetExtension
+from grandchallenge.core.utils.markdown import (
+    EmbedYoutubeExtension,
+    LinkBlankTargetExtension,
+)
 
 register = template.Library()
 
 
+def allowed_iframe(tag, name, value):
+    """Returns true if the iframe tag is allowed, false otherwise"""
+    if tag != "iframe":
+        return False
+    if name in settings.BLEACH_ALLOWED_FRAME_ATTRIBUTES:
+        return True
+    if name == "src":
+        p = urlparse(value)
+        if not p.scheme or not p.netloc:
+            return False
+        source = f"{p.scheme}://{p.netloc}"
+        if source in settings.BLEACH_ALLOWED_FRAME_SRC:
+            return True
+    return False
+
+
 @register.filter
-def clean(html: str):
+def clean(html: str, allow_iframes=False):
     """Clean the html with bleach."""
+
+    allowed_tags = settings.BLEACH_ALLOWED_TAGS
+    allowed_attributes = settings.BLEACH_ALLOWED_ATTRIBUTES
+
+    if allow_iframes:
+        allowed_tags = [*allowed_tags, "iframe"]
+        allowed_attributes.update({"iframe": allowed_iframe})
```

Comment:
This updates the setting as it's not a (deep)copy, right?

Reply:
Oh yes! Good find. Almost made the same error here as I fixed last time =P